packages/adapter-cloudflare-workers/index.js

```diff
@@ -1,73 +1,78 @@
-import fs from 'fs';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { relative } from 'path';
 import { execSync } from 'child_process';
 import esbuild from 'esbuild';
 import toml from '@iarna/toml';
 import { fileURLToPath } from 'url';
 
-/**
- * @typedef {import('esbuild').BuildOptions} BuildOptions
- */
-
 /** @type {import('.')} */
-export default function (options) {
+export default function () {
 	return {
 		name: '@sveltejs/adapter-cloudflare-workers',
 
-		async adapt({ utils }) {
-			const { site } = validate_config(utils);
+		async adapt(builder) {
+			const { site } = validate_config(builder);
 
 			const bucket = site.bucket;
 			const entrypoint = site['entry-point'] || 'workers-site';
 
 			const files = fileURLToPath(new URL('./files', import.meta.url));
+			const tmp = builder.getBuildDirectory('cloudflare-workers-tmp');
+
+			builder.rimraf(bucket);
+			builder.rimraf(entrypoint);
 
-			utils.rimraf(bucket);
-			utils.rimraf(entrypoint);
+			builder.log.info('Prerendering static pages...');
+			await builder.prerender({
+				dest: bucket
+			});
 
-			utils.log.info('Installing worker dependencies...');
-			utils.copy(`${files}/_package.json`, '.svelte-kit/cloudflare-workers/package.json');
+			builder.log.info('Installing worker dependencies...');
+			builder.copy(`${files}/_package.json`, `${tmp}/package.json`);
 
 			// TODO would be cool if we could make this step unnecessary somehow
-			const stdout = execSync('npm install', { cwd: '.svelte-kit/cloudflare-workers' });
-			utils.log.info(stdout.toString());
+			const stdout = execSync('npm install', { cwd: tmp });
+			builder.log.info(stdout.toString());
+
+			builder.log.minor('Generating worker...');
+			const relativePath = relative(tmp, builder.getServerDirectory());
+
+			builder.copy(`${files}/entry.js`, `${tmp}/entry.js`, {
+				replace: {
+					APP: `${relativePath}/app.js`
+				}
+			});
 
-			utils.log.minor('Generating worker...');
-			utils.copy(`${files}/entry.js`, '.svelte-kit/cloudflare-workers/entry.js');
+			writeFileSync(
+				`${tmp}/manifest.js`,
+				`export const manifest = ${builder.generateManifest({
+					relativePath
+				})};\n`
+			);
 
-			/** @type {BuildOptions} */
-			const default_options = {
-				entryPoints: ['.svelte-kit/cloudflare-workers/entry.js'],
+			await esbuild.build({
```

Comment:
It looks like cloudflare supports ES modules now, so maybe we can ditch `esbuild`?

https://blog.cloudflare.com/workers-javascript-modules/

Reply:
This doesn't appear to work for Cloudflare Pages yet — in wrangler I get this...

> SyntaxError: Cannot use import statement outside a module

...and on trying to deploy I get this:

> Deploying to Cloudflare's global network

Hopefully it'll get ironed out at some point, and we can drop the final bundling step in a future PR